src/redux/reducers/posts.js

```diff
@@ -0,0 +1,17 @@
+const initialState = { posts: [], loading: true }
+
+export default function posts(state = initialState, action) {
+  switch (action.type) {
+    case 'GET_POSTS_REJECTED':
+      return { ...state, posts: [], loading: false }
+
+    case 'GET_POSTS_SUCCESS':
+      return { ...state, posts: action.payload, loading: false }
+
+    case 'GET_POSTS_REQUESTED':
+      return { state }
```

Comment:
so whenever `'GET_POSTS_REQUESTED'` happens. You're going to be loading data whilst you wait for a success or rejected to happen. You also might want to set posts back to `[]`.

(This is the most common redux code we write, and that's how we tend to do it!)